README.md

```diff
@@ -86,6 +86,34 @@ The cache uses `catbox` to provide a simple pluggable interface, this supports s
 * http-transport:{version}:response - Basic response from a call cached for the duration of the `max-age` value key on just the URL of the response.
 * http-transport:{version}:staleResponse - Stale response from a called cached for the `stale-if-error` value keyed on just the URL of the response.
 
+Additionally, cache keys can be configured by passing a `varyOn` option. `varyOn` should be an array containing values that you want to vary on. For example, let's say you make requests to the same endpoint but values of some headers vary. By letting `http-transport-cache` know which headers vary, it will first check whether the headers actually exist in the request. If they do then it will construct a unique cache key which contains headers and their values.
+
+Example:
+
+We make a `GET` request to the following URL: `www.example.com/some-cacheable-path`.
+
+We vary on `accept-language` and `accept`. These headers will exist in the request. We pass in `varyOn` together with other options to configure the plugin.
+
+```js
+const opts = {
+  timeout: 2000,
+  varyOn: [
+    'accept-language',
+    'accept'
+  ]
+};
+```
+
+On the first request, the value of `accept-language` is `en` and `accept` is `application/json`. The resulting key will be:
+
+* GET:www.example.com/some-cacheable-path:accept-language=en,accept=application/json
```

Comment:
Is there a case for putting the header part of the key before the URL?
`www.example.com/some-cacheable-path:accept-language=en,accept=application/json` would also be a valid URL, wouldn't it? A GET to `http://www.example.com/some-cacheable-path:accept-language=en,accept=application/json` without any cache headers would result in a duplicate key.

Whereas `accept-language=en,accept=application/json:www.example.com/some-cacheable-path:accept-language=en,accept=application/json` is not a valid URL I believe so there would be no risk of collision.

Reply:
At the moment there isn't a case for putting the header part of the key before the URL.

If there are no headers in the request and we don't provide `varyOn` option then yes it would result in a duplicate key. 

But would't this mean that the response will be the same if we are making a `GET` request to the same URL multiple times?

Reply:
Let's walk this through.
My opts:
```js
const opts = {
  timeout: 2000,
  varyOn: [
    'accept-language',
    'accept'
  ]
};
```
1. Make request `GET accept-language=en, accept=json http://myurl.com/a`
Response is "foo"
Cache key becomes `GET:myurl.com/a:accept-language=en,accept=json`

2. Make very different request `GET http://myurl.com/a:accept-language=en,accept=json`
Response should be "bar"
Cache key becomes `GET:myurl.com/a:accept-language=en,accept=json`
 -> Cache key matches the previous request
 -> I get "bar" when I expect foo

Reply:
Yeah, I see what you mean and in this case the key will be identical. Good point. How high are the chances of this URL structure?

Reply:
Actually, following your other comment, I just realised that the cache key for request 2 would be `GET:myurl.com/a:accept-language=en,accept=json:accept-language=,accept=` so this may not be an issue.

Reply:
Ah yeah, makes sense.